src/apps/vpn/controller.cpp

```diff
@@ -234,32 +234,39 @@ bool Controller::activate(const ServerData& serverData,
     request->get(AppConstants::apiUrl(AppConstants::Account));
 
     connect(request, &NetworkRequest::requestFailed, this,
-            [this](QNetworkReply::NetworkError error, const QByteArray&) {
+            [this, serverSelectionPolicy](QNetworkReply::NetworkError error,
+                                          const QByteArray&) {
               logger.error() << "Account request failed" << error;
               REPORTNETWORKERROR(error, ErrorHandler::DoNotPropagateError,
                                  "PreActivationSubscriptionCheck");
 
               // Check if the error propagation has changed the Mozilla VPN
               // state. Continue only if the user is still authenticated and
               // subscribed.
-              if (MozillaVPN::instance()->state() == MozillaVPN::StateMain) {
-                setState(StateConnecting);
-              } else {
-                setState(StateOff);
+              if (MozillaVPN::instance()->state() != MozillaVPN::StateMain) {
+                return;
```

Comment:
This is a `bool` method, why isn't this `return false;`?

Reply:
This leaves us in state `StateCheckSubscription`, is that what we want?

Reply:
Do we have docs for this state machine?

Reply:
1. this return is for the nested function, which does not have a return value. So, `return` is correct
2. No because of this: https://github.com/mozilla-mobile/mozilla-vpn-client/blob/main/src/apps/vpn/mozillavpn.cpp#L139-L146 - when MozillaVPN state id != StateMain, the controller is deactivated automatically
3. not enough. Basically not yet.

Reply:
> this return is for the nested function, which does not have a return value. So, return is correct

doh. thanks.